name: Pull Request Check Workflow
on:
  workflow_call:

jobs:
  commitlint:
    runs-on: ubuntu-latest
    steps:
      - name: Checkout
        uses: actions/checkout@v3
        with:
          fetch-depth: 0
      - name: Commitlint
        uses: wagoid/commitlint-github-action@v5
        id: commitlint-github-action
        with:
          configFile: commitlint.config.js
          failOnWarnings: true
          helpURL: https://adup-tech.atlassian.net/wiki/spaces/UA/pages/3509190661/Git+Commit+Message+Convention
        env:
          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
      - name: Comment PR
        if: ${{ failure() }}
        uses: thollander/actions-comment-pull-request@v1
        with:
          message: |
            @${{ github.event.pull_request.user.login }}
            #### Commit linter errors/warnings:

            commit message: *${{ fromJSON(steps.commitlint-github-action.outputs.results)[0].message }}*

            :warning:
            > ${{ join(fromJSON(steps.commitlint-github-action.outputs.results)[0].errors, '<br>') }}

            > ${{ join(fromJSON(steps.commitlint-github-action.outputs.results)[0].warnings, '<br>') }}

            Please fix it!
            see: [Commit message conventions](https://adup-tech.atlassian.net/wiki/spaces/UA/pages/3509190661/Git+Commit+Message+Convention)
          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}